op-challenger/cmd/create_game.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/ethereum-optimism/optimism/op-challenger/flags"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr/metrics"
+	"github.com/urfave/cli/v2"
+)
+
+const envVarPrefix = "OP_CHALLENGER"
+
+var (
+	TraceTypeFlag = &cli.StringFlag{
+		Name:    "trace-type",
+		Usage:   "Trace types to support.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "TRACE_TYPE"),
+		Value:   config.TraceTypeCannon.String(),
+	}
+	OutputRootFlag = &cli.StringFlag{
+		Name:    "output-root",
+		Usage:   "The output root for the fault dispute game.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "OUTPUT_ROOT"),
+	}
+	L2BlockNumFlag = &cli.StringFlag{
+		Name:    "l2-block-num",
+		Usage:   "The l2 block number for the game.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "L2_BLOCK_NUM"),
+	}
+)
+
+func CreateGame(ctx *cli.Context) error {
+	logger, err := setupLogging(ctx)
+	if err != nil {
+		return err
+	}
+	rpcUrl := ctx.String(flags.L1EthRpcFlag.Name)
+	if rpcUrl == "" {
+		return fmt.Errorf("missing %v", flags.L1EthRpcFlag.Name)
+	}
+	gameAddr, err := opservice.ParseAddress(ctx.String(GameAddressFlag.Name))
+	if err != nil {
+		return err
+	}
+	outputRoot, err := opservice.ParseAddress(ctx.String(OutputRootFlag.Name))
+	if err != nil {
+		return err
+	}
+	traceType := ctx.Uint64(TraceTypeFlag.Name)
+	l2BlockNum := ctx.Uint64(L2BlockNumFlag.Name)
+
+	l1Client, err := dial.DialEthClientWithTimeout(ctx.Context, dial.DefaultDialTimeout, logger, rpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to dial L1: %w", err)
+	}
+	defer l1Client.Close()
+
+	caller := batching.NewMultiCaller(l1Client.Client(), batching.DefaultBatchSize)
+	contract, err := contracts.NewFaultDisputeGameContract(gameAddr, caller)
+	if err != nil {
+		return fmt.Errorf("failed to create dispute game bindings: %w", err)
+	}
+
+	txMgrConfig := txmgr.ReadCLIConfig(ctx)
+	txMgr, err := txmgr.NewSimpleTxManager("challenger", logger, &metrics.NoopTxMetrics{}, txMgrConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create the transaction manager: %w", err)
+	}
+
+	inner := contract.Inner()
+	methodCreateGame := "create"
+
+	call := inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	_, err = caller.SingleCall(context.Background(), batching.BlockLatest, call)
+	if err != nil {
+		return fmt.Errorf("failed to call create: %w", err)
+	}
+
+	call = inner.Call(methodCreateGame, uint32(traceType), outputRoot, l2BlockNum)
+	txCandidate, err := call.ToTxCandidate()
+	if err != nil {
+		return fmt.Errorf("failed to create tx candidate: %w", err)
+	}
```

Comment:
Why are we simulating the call first before actually sending the tx?

Reply:
The tx won't actually be sent because `eth_estimateGas` fails if the EVM reverts so I don't _think_ the extra call buys you anything.  But it might be worth testing to see if it does give a better user experience in some way (e.g the txmgr may keep retrying to send the tx for quite a while).  If its useful I'm happy to keep it, just figured it was worth removing if it doesn't actually give us anything.

Reply:
Got it ok, I'll remove then to prefer simplicity over verbosity in this case.